pkg/apis/extensions/validation/validation.go

```diff
@@ -345,25 +345,29 @@ func ValidateJobSpec(spec *extensions.JobSpec) errs.ValidationErrorList {
 	if spec.Completions != nil && *spec.Completions < 0 {
 		allErrs = append(allErrs, errs.NewFieldInvalid("completions", spec.Completions, isNegativeErrorMsg))
 	}
-
-	selector := labels.Set(spec.Selector).AsSelector()
-	if selector.Empty() {
+	if spec.Selector == nil {
 		allErrs = append(allErrs, errs.NewFieldRequired("selector"))
+	} else {
+		allErrs = append(allErrs, ValidatePodSelector(spec.Selector).Prefix("selector")...)
 	}
 
 	if spec.Template == nil {
 		allErrs = append(allErrs, errs.NewFieldRequired("template"))
-	} else {
+		return allErrs
+	}
+
+	if selector, err := extensions.PodSelectorAsSelector(spec.Selector); err == nil {
 		labels := labels.Set(spec.Template.Labels)
 		if !selector.Matches(labels) {
 			allErrs = append(allErrs, errs.NewFieldInvalid("template.labels", spec.Template.Labels, "selector does not match template"))
```

Comment:
Added metadata but i think spec should be added here https://github.com/mikedanese/kubernetes/blob/dne/pkg/apis/extensions/validation/validation.go#L335